components/scheduler2/util.go

```diff
@@ -0,0 +1,57 @@
+package scheduler2
+
+import (
+	"fmt"
+	"time"
+
+	"go.temporal.io/server/service/history/consts"
+	"go.temporal.io/server/service/history/hsm"
+)
+
+// ValidateTask ensures that the given transition is possible with the current
+// state machine state.
+func ValidateTask[
```

Comment:
I think this will generally be applicable but you should consider renaming to something that's more specific. This won't validate any task, just validate that a transition is valid for the given state.